### PR TITLE
Use absolute target names in sqlite3

### PR DESCRIPTION
**Pull request by tapia**
Conan feature related: https://github.com/conan-io/conan/pull/10099

### CODE REVIEW

File: recipes/sqlite3/all/conanfile.py [hunk 1]
```diff
@@ -174,10 +174,10 @@ def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "SQLite3")
```
Comment (SpaceIm) [inline recipes/sqlite3/all/conanfile.py:174]:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "SQLite3")
        self.cpp_info.set_property("cmake_target_name", "SQLite::SQLite3")")
```

File: recipes/sqlite3/all/conanfile.py [hunk 2]
```diff
@@ -174,10 +174,10 @@ def package_info(self):
...
         self.cpp_info.names["cmake_find_package"] = "SQLite"
         self.cpp_info.names["cmake_find_package_multi"] = "SQLite"
-        self.cpp_info.set_property("cmake_target_name", "SQLite")
+        self.cpp_info.set_property("cmake_target_name", "SQLite::SQLite")
```
Comment (SpaceIm) [inline recipes/sqlite3/all/conanfile.py:177]:
```suggestion
```

File: recipes/sqlite3/all/conanfile.py [hunk 3]
```diff
@@ -174,10 +174,10 @@ def package_info(self):
...
         self.cpp_info.components["sqlite"].names["cmake_find_package"] = "SQLite3"
         self.cpp_info.components["sqlite"].names["cmake_find_package_multi"] = "SQLite3"
-        self.cpp_info.components["sqlite"].set_property("cmake_target_name", "SQLite3")
+        self.cpp_info.components["sqlite"].set_property("cmake_target_name", "SQLite::SQLite3")
```
Comment (SpaceIm) [inline recipes/sqlite3/all/conanfile.py:180]:
```suggestion
```

File: recipes/sqlite3/all/conanfile.py [hunk 4]
```diff
@@ -174,10 +174,10 @@ def package_info(self):
...
         self.cpp_info.components["sqlite"].builddirs.append(self._module_subfolder)
         self.cpp_info.components["sqlite"].build_modules["cmake_find_package"] = [self._module_file_rel_path]
```
Comment (SpaceIm) [inline recipes/sqlite3/all/conanfile.py:182]: how tu support in CMakeDeps these custom CMake variables defined in this module?

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/sqlite3/all/conanfile.py (hunk 4 above)
Comment (czoido) [reply recipes/sqlite3/all/conanfile.py:182]: I guess we can add
```
self.cpp_info.set_property("cmake_build_modules", [build_module_for_cmakedeps])
```

File: recipes/sqlite3/all/conanfile.py [hunk 5]
```diff
@@ -172,12 +172,12 @@ def package_info(self):
         self.cpp_info.filenames["cmake_find_package"] = "SQLite3"
         self.cpp_info.filenames["cmake_find_package_multi"] = "SQLite3"
         self.cpp_info.set_property("cmake_file_name", "SQLite3")
+        self.cpp_info.set_property("cmake_target_name", "SQLite::SQLite3")
```
Comment (SpaceIm) [inline recipes/sqlite3/all/conanfile.py:175]:
```suggestion
        self.cpp_info.set_property("cmake_target_name", "SQLite::SQLite3")
        self.cpp_info.set_property("cmake_find_mode", "both") 
```

https://cmake.org/cmake/help/latest/module/FindSQLite3.html

File: recipes/sqlite3/all/conanfile.py (hunk 4 above)
Comment (SpaceIm) [reply recipes/sqlite3/all/conanfile.py:183]: So yes maybe adding `self.cpp_info.set_property("cmake_build_modules", [self._module_file_rel_path])` works with `self.cpp_info.set_property("cmake_find_mode", "both")`

Comment (czoido) [reply recipes/sqlite3/all/conanfile.py:183]: It has just been included in https://github.com/conan-io/conan/pull/10227

**Review by czoido — DISMISSED** [review]

**Review by czoido — APPROVED** [review]

File: recipes/sqlite3/all/conanfile.py [hunk 6]
```diff
@@ -172,12 +172,14 @@ def package_info(self):
...
+        self.cpp_info.set_property("cmake_find_mode", "both") 
         self.cpp_info.names["cmake_find_package"] = "SQLite"
         self.cpp_info.names["cmake_find_package_multi"] = "SQLite"
-        self.cpp_info.set_property("cmake_target_name", "SQLite")
         self.cpp_info.components["sqlite"].names["cmake_find_package"] = "SQLite3"
         self.cpp_info.components["sqlite"].names["cmake_find_package_multi"] = "SQLite3"
-        self.cpp_info.components["sqlite"].set_property("cmake_target_name", "SQLite3")
+        self.cpp_info.components["sqlite"].set_property("cmake_target_name", "SQLite::SQLite3")
+        self.cpp_info.components["sqlite"].set_property("cmake_build_modules", [self._module_file_rel_path])
```
Comment (SpaceIm) [inline recipes/sqlite3/all/conanfile.py:182]:
```suggestion
```

Not required with https://github.com/conan-io/conan/pull/10227

**Review by SpaceIm — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**